extensions/amp-fx-collection/0.1/providers/fade-in.js

```diff
@@ -0,0 +1,205 @@
+/**
```

Comment:
Haven't given a detailed view at this, but it looks to me like most of this can be abstracted out? Is there a reason why we can't describe the animations themselves in a more declarative way?

Reply:
I'm sorry I don't understand the question here, could you rephrase it?

Reply:
Sure.

It seems to me like the provider does the following:

1. Gathers configuration parameters and sets default values.
2. Calculates relative position for triggering, etc.
3. Applies interpolation (`update_`)

These semantics will apply to most, if not all, fx presets.

If we're writing a provider for each of them, we will have some duplicated code and it will probably be a little tough to maintain them.

My question really is whether something is blocking you from writing a generic `FxProvider` that simply takes in an interpolation function and/or a default installer. Similar to the API in `animation.js`.

Reply:
Ahhh gotcha. Can definitely do that! Thanks for the explanation.